### ISSUE TITLE
why is there no acceleration of plant growth?

### ISSUE THREAD

**Issue by lijspoop** [jodeks-datapacks/Sleep#4]
I may not have found the settings, but I think when the player goes to bed, there should be an acceleration of plant growth using the [randomTickSpeed](https://minecraft.fandom.com/wiki/Game_rule) rule, but this does not happen. please add this in the next update.

**Comment by Mqxx** [comment]
Hey, thanks for your feature request.

Unfortunately there are a few problems with this method that can't be solved with `randomTickSpeed`.

First of all, `randomTickSpeed` does not simply skip a certain time but only accelerates the time by the respective factor. Why is this a problem? The question is at what point of the sleep should the `randomTickSpeed` be increased? And for how long?

Let's assume we increase the `randomTickSpeed` at the end when it becomes day again. Then all the plants would need a certain amount of time to grow. In other words, when the player gets up, he sees plants and trees shooting out of the ground all at once. And that's not very nice/realistic.

Another possibility would be to increase the `randomTickSpeed` while the player is still in bed. The problem with this, however, is that the player could then very strongly abuse this behavior by repeatedly going to bed and getting up again. There is no way to force the player to stay in bed. This also means that it is not possible to avoid this behavior.

To summarize, there are only two ways to implement this feature. Which of the two methods would you prefer? Or do you have another method to solve the problem?

**Comment by lijspoop** [comment]
The main focus is on the mechanics of accelerating growth, even if not "very nice" looking, however, the mechanics themselves are realistic. For the first way, a separate setting could be added to adjust how long the acceleration lasts. In the second way, the existing "time warp" implementation (`/scoreboard players set &time_control sleep.config value`) can be used, but I don't know if it has the same abuse issues. I used this datapack on a server with 2-4 players, so I didn't think about abuse.

I would prefer to accelerate the growth of plants and not think about how good it looks.

**Comment by Jodekq** [comment]
I mean I could implement it with a setting and whoever wants this feature can activate it. What they do then e.g. abusing it is there problem then.

**Comment by lijspoop** [comment]
Wait, I think I might be misunderstanding what "abuse" you are referring to. Are you saying that a player could constantly go to bed and get up, thus abusing the feature? Could that really happen? After all, players need to lie down for a few seconds before the acceleration starts.

Maybe I'm still misunderstanding, but is it possible to try implementing a gradual increase in `randomTickSpeed`? I think this could help protect against abuse (as I understand it). Again, adding a corresponding setting would be beneficial.

I'm not very familiar with the nuances of `randomTickSpeed`. From the description on the wiki, I understood that the block tick is accelerated, and this can be used to speed up plant growth.

If you are referring to players being able to change the time to the next night every time the previous one ends, then that’s their problem with abuse, and it should be addressed by those who use the time-changing mechanic.

---

I don't have another solution to the problem of abusing this feature. So, if this feature is included in the next update, I would like to close this issue.

**Comment by Jodekq** [comment]
Yup, tho it isn't probably a big deal with the time acceleration mode, with the instant skip mode it isn't that easy.

If you want to make the mode "abuse proof" you'd need to put the randomtickspeed to a very high value just before you've slept threw the night. 

So this might a problem then, as it could be the reason gor lags.

But anyway it sounds like a really good idea and I'll try to implement in some way.
I'll keep you updated.

**Comment by Mqxx** [comment]
> If you want to make the mode "abuse proof" you'd need to put the randomtickspeed to a very high value just before you've slept threw the night.

This 👆

**Comment by Jodekq** [comment]
btw. got implemented some time ago
